include/flang/Common/enum-set.h

```diff
@@ -16,11 +16,11 @@
 
 #include "constexpr-bitset.h"
 #include "idioms.h"
+#include "llvm/Support/raw_ostream.h"
```

Comment:
We can actually forward declare `llvm::raw_ostream` (unlike `std::ostream`) here.
So instead of including the header you can do:
```
namespace llvm {
class raw_ostream;
}
```
and then include the header in the cpp file instead for faster compile times.

This also applies for most of the headers below.

Reply:
+1

Reply:
Ok.

Reply:
This header has this method llvm::raw_ostream &Dump implemented in enum-set, so the compile complains if I remove the header. 
It is ok to have header + namespace?
llvm::raw_ostream &Dump(
      llvm::raw_ostream &o, std::string EnumToString(enumerationType)) const {
    char sep{'{'};
    IterateOverMembers([&](auto e) {
      o << sep << EnumToString(e);
      sep = ',';
    });

Reply:
I think we should move this function definition to the corresponding cpp file instead and just leave the declaration here

Reply:
Should we do this in This PR or should I make another one? There are more header files that were implemented this way? We need to merge this before merging with LLVM.

Reply:
I think this should be quite a small change to this PR so you should just do it here in my opinion

Reply:
We think that these changes may lead to further changes. For now, I will leave this task for later as we have a deadline for this PR.